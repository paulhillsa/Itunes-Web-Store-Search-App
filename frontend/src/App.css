.App {
  text-align: center;
}

.table {
  padding: 0% !important;
  margin: 0% !important;
}

.cardsContainer {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-column-gap: 10px;
  grid-row-gap: 10px;
  padding: 150px;
  align-items: center;
}

.card {
  border-radius: 5px;
  height: 450px;
  text-align: center;
}

.star {
  position: absolute;
  bottom: 370px;
  left: 80px;
  margin: auto;
  width: 100%;
  height: 50px;
  color: gold;
  text-align: center;
  z-index: 1;
}

.card-img-top {
  width: 100%;
  height: 15vw;
  object-fit: cover;
}


.header {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-column-gap: 10px;
  grid-row-gap: 10px;
  padding: 25px;
  position: fixed;
  top: 0;
  left: 5%;
  width: 100%;
  z-index: 2;
  background-color: white;
}

.favouritesContainer {
  padding-bottom: 100px;
}


.tableContainer {
  padding: 5px !important;
}

tr, td {
  padding: 5px !important;
  width: 100px;
  height: auto;
}

#noResults {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 70vh;
  width: 120vh;
  font-size: 50px;
  font-weight: lighter;
  
}
