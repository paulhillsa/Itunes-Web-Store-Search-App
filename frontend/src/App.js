import React, {Component} from 'react';
import {BrowserRouter as Router, Routes, Route} from 'react-router-dom';
import './App.css';

import Home from './components/Home.js';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
    }
  }


  render() {
    return (
      <div className="App">
        {/* Bootstrap styling */}
        <link rel = "stylesheet"
        href = "https://maxcdn.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css"
        integrity = "sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T"
        crossorigin = "anonymous" />

        {/* Pathing */}
        <Router>
          <Routes>
            <Route path="/" element={<Home />} />
            <Route path="/search" element={<Home />} />
          </Routes>
        </Router>
      </div>
    )
  }
}


