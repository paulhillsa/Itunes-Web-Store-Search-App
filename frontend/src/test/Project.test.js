import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';
import handleSubmit from '../components/Search';
import axios from 'axios';

//Snapshot test for the App component
test('App renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
});

// Unit test for the Search component
describe('handleSubmit', () => {
    test('should return a function', () => {
        expect(typeof handleSubmit).toBe('function');
    })
})

// unit test for the axios call
describe('axios call', () => {
    test('should return a promise', () => {
        expect(axios.get).toBeDefined();
    });
}
);

// localhost:3001/search/red/podcast should return data
test('should return data', () => {
    expect.assertions(1);
    return axios.get('http://localhost:3001/search/red/podcast').then(response => {
        expect(response.data).toEqual(expect.any(Object));
    });
}
);

// sendng a bad request should return an error
test('should return an error', () => {
    expect.assertions(1);
    return axios.get('http://localhost:3001/search/red/podcassbjhjhgts').then(response => {
        expect(response.data).toEqual(expect.any(Object));
    });
}
);





