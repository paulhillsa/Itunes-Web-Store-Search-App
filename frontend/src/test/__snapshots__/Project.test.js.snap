// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="App"
>
  <link
    crossorigin="anonymous"
    href="https://maxcdn.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css"
    integrity="sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T"
    rel="stylesheet"
  />
  <div>
    <div
      className="container"
    >
      <div
        className="header"
      >
        <form
          className=""
          onChange={[Function]}
        >
          <input
            className="mr-sm-2 form-control"
            placeholder="Search term..."
            type="text"
          />
        </form>
        <div
          className="dropdown"
        >
          <button
            aria-expanded={false}
            className="dropdown-toggle btn btn-primary"
            disabled={false}
            id="dropdown-basic-button"
            onClick={[Function]}
            type="button"
          >
            Catagories
          </button>
        </div>
        <div
          className="searchButton"
        />
        <div
          className="helpMoidal"
        >
          <button
            className="btn btn-btn btn-secondary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
             Help 
          </button>
        </div>
      </div>
      <div
        className="cardsContainer container"
      >
        <div
          className="noResults"
        >
          <h1
            id="noResults"
          >
             Itunes Web Store Search
          </h1>
        </div>
      </div>
      <div
        className="favouritesContainer container"
      >
        <div
          className="noFavourites"
        />
      </div>
    </div>
  </div>
</div>
`;
